### PR TITLE
[TRA-434] Governance msg MsgSetMarketMapperRevShareDetailsForMarket

### PR DESCRIPTION
**Pull request by chenyaoy**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced new functionality to set market mapper revenue share details for specific markets. This includes new message types and RPC methods for managing revenue share settings.

- **Documentation**
  - Updated internal messages and test files to include new message types related to market revenue sharing.

### CODE REVIEW

File: protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+				Authority: lib.GovModuleAddress.String(),
+				Params: types.MarketRevShareDetailsParams{
+					MarketId: constants.MarketId0,
+					MarketMapperRevShareDetails: &types.MarketMapperRevShareDetails{
+						ExpirationTs: 100,
+					},
+				},
+			},
+			expectedErr: "",
+		},
```
Comment (shrenujb) [inline protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details_test.go:33]: Add failure cases for invalid authority and empty authority

File: protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	// Check if sender is authorized to set revenue share
+	if !k.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+
```
Comment (shrenujb) [inline protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go:27]: Need validation to check if the provided marketId is valid.

You can do this by fetching the market from the prices keeper and checking if it exists

File: proto/dydxprotocol/revshare/params.proto [hunk 3]
```diff
@@ -19,3 +20,13 @@ message MarketMapperRevenueShareParams {
   // the revenue share goes down to 0
   uint32 valid_days = 3;
 }
+
+// MarketRevShareDetailsParams represents params for the 
+// MsgSetMarketMapperRevShareDetailsForMarket message
+message MarketRevShareDetailsParams {
```
Comment (shrenujb) [inline proto/dydxprotocol/revshare/params.proto:26]: I would either get rid of this proto completely and just add the market_id to the top level struct or rename this to `SetMarketRevShareDetailsParams`

File: protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go (hunk 2 above)
Comment (chenyaoy) [reply protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go:27]: Chatted offline and decided not to have this validation as we assume governance is a strong safety check. Having this validation creates a circular dependency between x/prices and x/revshare

File: proto/dydxprotocol/revshare/params.proto (hunk 3 above)
Comment (chenyaoy) [reply proto/dydxprotocol/revshare/params.proto]: done!

File: protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details_test.go (hunk 1 above)
Comment (chenyaoy) [reply protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details_test.go:31]: done!

File: proto/dydxprotocol/revshare/tx.proto [hunk 4]
```diff
@@ -29,3 +36,21 @@ message MsgSetMarketMapperRevenueShare {
...
+// Msg to set market mapper revenue share details (e.g. expiration timestamp)
+// for a specific market. To be used as an override for existing revenue share
+// settings set by the MsgSetMarketMapperRevenueShare msg
+message MsgSetMarketMapperRevShareDetailsForMarket {
+  // The address that controls the module (the gov module account).
+  option (cosmos.msg.v1.signer) = "authority";
+
+  string authority = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+
+  uint32 market_id = 2;
```
Comment (shrenujb) [inline proto/dydxprotocol/revshare/tx.proto:49]: nit: add comment for this field

**Review by shrenujb — APPROVED** [review]

File: protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go (hunk 2 above)
Comment (vincentwschau) [reply protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go:27]: Call this out explicitly in a comment.

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**